readthedocs/core/unresolver.py

```diff
@@ -551,13 +551,23 @@ def unresolve_domain(self, domain):
         Unresolve domain by extracting relevant information from it.
 
         :param str domain: Domain to extract the information from.
+         It can be a full URL, in that case, only the domain is used.
         :returns: A UnresolvedDomain object.
         """
+        parsed_domain = urlparse(domain)
+        if parsed_domain.scheme:
+            if parsed_domain.scheme not in ["http", "https"]:
+                raise InvalidSchemeError(parsed_domain.scheme)
+            domain = parsed_domain.hostname
```

Comment:
Don't we need to call `self.domain_from_host` here?

Reply:
I guess you mean `get_domain_from_host`. We don't need to call it here, `hostname` returns the domain without the port already.